tests/cephfs/cephfs_mirroring/test_cephfs_mirror_auth_caps.py

```diff
@@ -0,0 +1,522 @@
+            target_fs,
+            target_user,
+            target_clients,
+            source_fs,
+            source_clients,
+            target_site_name,
+            permissions="rwp",
+        )
+        log.info("Secenario 2: Update user auths and try to create bootstrap token")
+        log.info("Create a user on target cluster for peer connection")
+        target_user_for_peer = fs_mirroring_utils.create_authorize_user(
+            target_fs, target_user, target_clients[0]
+        )
+        if target_user_for_peer:
+            log.error("User Creation Failed with the expected caps")
+            raise CommandFailed("User Creation Failed")
+
+        log.info(f"Enable cephfs mirroring module on the {source_fs}")
+        fs_mirroring_utils.enable_snapshot_mirroring(source_fs, source_clients[0])
+
+        log.info("Create the peer bootstrap")
+        bootstrap_token = fs_mirroring_utils.create_peer_bootstrap(
+            target_fs, target_user, target_site_name, target_clients[0]
+        )
+        out, rc = target_clients[0].exec_command(
+            sudo=True, cmd=f"ceph auth get client.{target_user}"
+        )
+        log.info(out)
+        log.info("Before Modifying the caps")
+        caps_list = ["r", "rw", "rwp"]
+        for caps in caps_list:
+            target_clients[0].exec_command(
+                sudo=True,
+                cmd=f'ceph auth caps client.{target_user} mds "allow {caps} fsname={target_fs}" '
+                f'mon "allow r fsname={target_fs}" '
+                f'osd "allow rw tag cephfs data={target_fs}"',
+            )
+            out, rc = target_clients[0].exec_command(
+                sudo=True, cmd=f"ceph auth get client.{target_user}"
+            )
+            log.info(out)
+            command = (
+                f"ceph fs snapshot mirror peer_bootstrap create "
+                f"{target_fs} client.{target_user} {target_site_name} -f json"
+            )
+            out, rc = target_clients[0].exec_command(
+                sudo=True, cmd=command, check_ec=False
+            )
+
+            if not rc:
+                log.error(f"Bootstrap token has been created with permissions {caps}")
+                raise CommandFailed(
+                    f"Bootstrap token has been created with permissions {caps} after updating user caps"
+                )
+
+        log.info("Import the bootstrap on source")
+        fs_mirroring_utils.import_peer_bootstrap(
+            source_fs, bootstrap_token, source_clients[0]
+        )
+
+        log.info("Get Peer Connection Information")
+        validate_peer_connection = fs_mirroring_utils.validate_peer_connection(
+            source_clients[0], source_fs, target_site_name, target_user, target_fs
+        )
+        if validate_peer_connection:
+            log.error("Peer Connection not established")
+            raise CommandFailed("Peer Connection failed to establish")
+
+        log.info("Create Subvolumes for adding Data")
```

Comment:
Can we have a generic method for adding data in case of mirroring tests? I see the same steps repeated in every other test of mirroring.

Reply:
Common functions already exist which help in configuring the mirroring.
If we automate end-to-end mirroring validation. it might create difficulties when we want to break any of the steps in it